src/app/containers/Article/index.jsx

```diff
@@ -48,43 +84,42 @@ class ArticleContainer extends Component {
       const data = await response.json();
       const amp = isAmpPath(path);
 
-      return { amp, data };
+      return { amp, data, service };
     } catch (error) {
       console.log(error); // eslint-disable-line no-console
       return {};
     }
   }
 
-  render() {
-    const { amp, data } = this.props;
-    /*
-      This handles async data fetching, and a 'loading state', which we should look to handle more intelligently.
+  /*
+    [1] This handles async data fetching, and a 'loading state', which we should look to handle more intelligently.
       After.JS gives no explicit loading state, so we just have to check for a lack of data.
       After.JS should handle async data fetching in a more informative way, which will be become an issue for error handling.
-    */
+  */
+  render() {
+    const { amp, data, service } = this.props;
     if (!data) {
-      return 'Loading...';
+      return 'Loading...'; /* [1] */
     }
-
     const { content, metadata, promo } = data;
-    const { id: aresArticleId } = metadata;
-
+    const { id: aresArticleId, tags } = metadata;
     const id = aresArticleId.split(':').pop();
-    const { blocks } = content.model;
-    const canonicalLink = `https://www.bbc.com/news/articles/${id}`;
-
-    const metadataProps = {
-      amp,
-      canonicalLink,
-      lang: metadata.passport.language,
-      title: promo.headlines.seoHeadline,
-    };
-
+    const config = serviceConfig[service];
     return (
       <Fragment>
         <Header />
-        <Metadata {...metadataProps} />
-        <MainContent blocks={blocks} />
+        <Metadata
+          {...metadataProps(
```

Comment:
@bbc/article-devs-news 

This component, especially it's `render` lifecycle, will be doing quite a bit of data handling when we abstract metadata.

We data fetch in `getInitialProps`, and do data abstraction/rendering in `render`. It is generally fine to do some level of data abstraction in the `render` lifecycle, but it should not be a place where we do substantial forming of a data model, or data fetching (getting config for a service could be data fetching).

It might make more sense to do all data fetching and abstraction (including config) in a single place, `getInitialProps`, and just pass that data to the component/`render`. This might make more sense as an application design pattern; `getInitialProps` is the source of truth for generating a model of data, and `render` just deals with rendering something based on the data that it is passed.

Further to this, what do you think about the concept of a data abstraction/config fetching layer, where in `getInitialProps` we just pass in `data` and we get back an object that includes the [`metadata`](https://github.com/bbc/simorgh/pull/647/files#diff-d5c925f82e0315a863ea6a24dd59cb60R33) and [`content`/`blocks`](https://github.com/bbc/simorgh/pull/647/files#diff-d5c925f82e0315a863ea6a24dd59cb60R122).

This would all create a succinct separation of concerns between data-fetching, data abstraction, and rendering.

What are people's thoughts? 

If this is something we want to do, we might want to do this layer before we do metadata as it could just be a duplication of effort, alternatively we could continue with this PR and make the changes afterwards.

Reply:
This is an example of what I refer to: #659

Reply:
I agree that the data manipulation should be happening in `getInitialProps` and then the props should be passed to the render method. Thanks for this review!

Reply:
I've just opened another issue to carry out this refactor: https://github.com/bbc/simorgh/issues/662